syft/pkg/cataloger/javascript/parse_node_binary.go

```diff
@@ -0,0 +1,38 @@
+package javascript
+
+import (
+	"regexp"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/anchore/syft/syft/source"
+)
+
+var nodeLookup = generic.Lookup{
+	Package: "node.js", // Note: this purposely matches the "node.js" string to aid nvd vuln matching
+	FilepathPatterns: []*regexp.Regexp{
+		// note: should we just parse all files resolved with executable mimetypes
+		// regexp that matches node binary
+		regexp.MustCompile(`(.*/|^)node$`),
+	},
+	EvidencePatternTemplates: []string{
+		// regex that matches node.js/vx.y.z
+		`(?m)node\.js\/v(?P<version>[0-9]+\.[0-9]+\.[0-9]+)`,
+	},
+}
+
+func parseNodeBinary(_ source.FileResolver, _ *generic.Environment, reader source.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	p, _, err := nodeLookup.Find(reader)
+	if err != nil {
+		log.Trace("failed to find node.js package: %+v", err)
+		return nil, nil, nil // we can silently fail here to reduce warning noise
+	}
+	// TODO add node specific metadata to the packages to help with vulnerability matching
+	if p != nil {
+		p.Language = pkg.JavaScript
```

Comment:
Is this accurate? As noted before, this is predominantly C code with some javascript... but perhaps it matches things in the ecosystem in GHSA/etc.? Without digging in to this I would think the `javascript` ecosystem would be for javascript packages but _not_ the node binaries

Reply:
^ The language might be a little loose - we could default this to `Unknown`, but if we're able to enter a namespace in the grype db where `node.js` has a vulnerability filed against it that was not caught or listed incorrectly in NVD this gives us a bit more of a shot at this package being compared against the correct data. I do agree it's awkward.

Reply:
I'm definitely on board with `pkg.JavaScript` being better than `pkg.Unknown`, if that's the only other option 👍

Reply:
That was for the pkg.Type which is being set in `Find` as `pkg.Binary`

`Find` sets the language as unkown and then leaves the decision to the caller on if want's to enhance it with a known language or ecosystem. Definitely awkward